Sources/Common/Core/ScalarsToColors/index.js

```diff
@@ -125,25 +125,26 @@ function vtkScalarsToColors(publicAPI, model) {
       newColors[count] = l;
       newColors[count + 1] = l;
       newColors[count + 2] = l;
-      newColors[count + 3] = convtFun(values[i + 1]);
+      newColors[count + 3] = convtFun(values[i + 1]) * alpha;
       count += 4;
     }
   };
 
   publicAPI.rGBToRGBA = (newColors, colors, alpha, convtFun) => {
-    const a = convtFun(alpha);
+    const a = floatColorToUChar(alpha);
 
     const values = colors.getData();
+    const newValues = newColors.getData();
     const size = values.length;
     const component = 0;
     const tuple = 3;
 
     let count = 0;
     for (let i = component; i < size; i += tuple) {
-      newColors[(count * 4)] = convtFun(values[i]);
-      newColors[(count * 4) + 1] = convtFun(values[i + 1]);
-      newColors[(count * 4) + 2] = convtFun(values[i + 2]);
-      newColors[(count * 4) + 3] = a;
+      newValues[(count * 4)] = convtFun(values[i]);
+      newValues[(count * 4) + 1] = convtFun(values[i + 1]);
+      newValues[(count * 4) + 2] = convtFun(values[i + 2]);
+      newValues[(count * 4) + 3] = a;
       count++;
```

Comment:
Why operate on newValues as opposed to newColors? The method above is still using newColors so I am unsure what is behind the change.